machine-runner/src/runner/runner.ts

```diff
@@ -531,7 +537,7 @@ export const createMachineRunnerInternal = <
     isDestroyed: internals.destruction.isDestroyed,
     noAutoDestroy: () =>
       MachineRunnerIterableIterator.make({
-        nextValueAwaiter,
+        nextValueAwaiter: nextValueAwaiter(defaultNextValueAwaiter.state()),
```

Comment:
alternatively we could also use the `.get()` value here — this would mean that a `.noAutoDestroy()` AsyncIterable always immediately yields the current state if there is one, then waits for the next update